src/gsi/obsmod.F90

```diff
@@ -1038,10 +1038,15 @@ subroutine init_directories(in_pe,num_pe)
              write(loc_pe_name,'(i4.4)') i
              loc_dirname = 'dir.'//trim(loc_pe_name)
 #ifdef __INTEL_COMPILER
-             l_mkdir_stat = MAKEDIRQQ(trim(loc_dirname))
-             if(.not. l_mkdir_stat) then
-                write(6, *) "Failed to create directory ", trim(loc_dirname), " for PE ", loc_pe_name
-                call stop2(678)
+             INQUIRE(directory=trim(loc_dirname), exist=l_dir_exist)
+             if (.not.l_dir_exist) then
+                l_mkdir_stat = MAKEDIRQQ(trim(loc_dirname))
+                if(.not.l_mkdir_stat) then
+                   ierror=GETLASTERRORQQ()
+                   write(6, *) "INIT_DIRECTORIES:  ***ERROR** Failed to create directory ", &
+                        trim(loc_dirname)," for PE ", loc_pe_name, ' ierror= '
```

Comment:
Missing the variable `ierror` in the write statement.
```suggestion
                        trim(loc_dirname)," for PE ", loc_pe_name, ' ierror= ', ierror
```

Reply:
Thank you, @DavidHuber-NOAA , for catching my mistake.  What's the point of capturing an error flag if you don't actually print the value of the flag when an error occurs.   Bug corrected at 2bc4f13f.